### DISCUSSION TITLE
Customizable Sleep Timer via App

### DISCUSSION THREAD

**Discussion by MichaelSeitz98** [bk1285/rpi_wordclock#257] [General]
Has someone already considered making the sleep mode times adjustable via the app/web interface? Would be quite cool to set it change it :) Are there any showstoppers? If not what steps/adjustments would be required for a such a development? 

Thanks a lot for your thoughts! 
Best regards 
Michael

**Comment by FrankX0** [comment]
Hi Michael,

This certainly would be possible.
This requires additions to the API and app.
I can have look into this, unless of course you want to do this yourself.

Regards,
Frank.

**Reply by MichaelSeitz98** [reply]
Hi Frank, 

Sure! I would appreciate it a lot! I will take a close look on your changes, maybe then I can learn from it for potential further extensions (as I am quite new to open-source dev)! 

Btw: very nice work with your Interface/App but also general here in the community 🙌

Thanks a lot already! 
Michael

**Comment by FrankX0** [comment]
Hi Michael,

Pull request #240 now also contains setting the sleep timer through the web-app:
![image](https://github.com/bk1285/rpi_wordclock/assets/7318792/de607df1-c680-420d-9c83-6d5339d76e80)
Please let me know if you find any defects (did not test it on an actual clock).

Regards,
Frank.

**Reply by MichaelSeitz98** [reply]
Hi Frank! 

Nice! Great work! I will check out your branch and test it on the actual watch next weekend. 

Thanks and all the best 
Michael

**Comment by MichaelSeitz98** [comment]
Hi @FrankX0, 

i did test it on my watch...but I do not get it to work as I get such a Error `OSError: [Errno 98] Address already in use
` Error. 

Do you have an idea or is it even related to my installation? 

[error_message_full.txt.txt](https://github.com/bk1285/rpi_wordclock/files/14543460/error_message_full.txt.txt)


Thanks again! 
Michael

**Reply by FrankX0** [reply]
This error message means it cannot open the port (80) which is used by the web-app. Do you have the original clock still running?

**Reply by MichaelSeitz98** [reply]
Thanks for the super fast reply. The original version should not be running, at least not that I know. I also replaced my crontab with wordclock.py of your branch. So the old version should not be running anymore :)

**Reply by FrankX0** [reply]
So, you replaced the crontab and then rebooted the pi?

**Reply by MichaelSeitz98** [reply]
Hi Frank, 
now it seem so work. Another reboot made it work! Thanks a lot.
